### ISSUE TITLE
Issue when running terraform plan with -target=module.x when multiple modules are used

### ISSUE THREAD

**Issue by ChristopheSchmitz** [hashicorp/terraform#11166]
Hello

### Terraform Version
Terraform v0.8.3

### Affected Resource(s)
Terraform's core

### Terraform Configuration Files
The structure looks like
```
├── main.tf
├── terraform.tfvars
├── variables.tf
└── vpc
    ├── variables.tf
    └── vpc.tf
```

main.tf Note that to reproduce the bug, I need to use two modules: a **and** b
```hcl
module a {
  source = "./vpc"
  region = "${var.region}"
  access_key = "${var.access_key}"
  secret_key = "${var.secret_key}"
  cidr_block = "10.1.0.0/16"
}
module b {
  source = "./vpc"
  region = "${var.region}"
  access_key = "${var.access_key}"
  secret_key = "${var.secret_key}"
  cidr_block = "10.2.0.0/16"
}

```
vpc.tf
```hcl
provider aws {
  region = "${var.region}"
  access_key = "${var.access_key}"
  secret_key = "${var.secret_key}"
}
resource aws_vpc my_vpc {
  cidr_block = "${var.cidr_block}"
}
```

### Expected Behavior
The module specified in the target should be planed correctly

### Actual Behavior
Error message:
```
Error running plan: 1 error(s) occurred:

* 1:3: unknown variable accessed: var.secret_key in:

${var.secret_key}
```
### Steps to Reproduce
terraform plan -target=module.a

Note that if I remove the definition of `module b` in main.tf, then it works as intended.
Note that using `-Xlegacy-graph` is a work around

### References
- GH-10911 Looks similar, but it is different as GH-10911 mention that `terraform plan` works.

**Comment by mcanevet** [comment]
Related to https://github.com/hashicorp/terraform/issues/10711 and https://github.com/hashicorp/terraform/issues/10911

**Comment by mitchellh** [comment]
This should be fixed by #11373 and friends coming in 0.9. The issue is the old graphs being used for input.

**Comment by mitchellh** [comment]
Fixed!
